File: android/app/src/main/java/uk/co/amlcurran/social/EventViewHolder.java
package uk.co.amlcurran.social;

import android.view.View;
import android.widget.TextView;

import org.joda.time.format.DateTimeFormat;
import org.joda.time.format.DateTimeFormatter;

public class EventViewHolder extends CalendarItemViewHolder<EventCalendarItem> {

    private final TextView textView;
    private final WhatsOnAdapter.EventSelectedListener eventSelectedListener;
    private final TextView subtitle;
    private final DateTimeFormatter formatter = DateTimeFormat.shortTime();
    private final JodaCalculator jodaCalculator = new JodaCalculator();

    public EventViewHolder(View itemView, WhatsOnAdapter.EventSelectedListener eventSelectedListener) {
        super(itemView);
        this.eventSelectedListener = eventSelectedListener;
        this.textView = itemView.findViewById(R.id.textView);
        this.subtitle = itemView.findViewById(R.id.event_subtitle);
    }

    @Override
    public void bind(final EventCalendarItem item) {
        textView.setText(item.getTitle());
        subtitle.setText(subtitle.getResources().getString(R.string.event_from, formatter.print(jodaCalculator.getDateTime(item.getStartTime()))));
        itemView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                eventSelectedListener.eventSelected(item, itemView);
            }
        });
    }

}

File: android/app/src/main/java/uk/co/amlcurran/social/EventViewHolder.kt
package uk.co.amlcurran.social

import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.ui.Modifier
import androidx.compose.ui.platform.ComposeView
import org.joda.time.DateTime

import org.joda.time.format.DateTimeFormatter

internal class EventViewHolder(private val composeView: ComposeView, private val eventSelectedListener: WhatsOnAdapter.EventSelectedListener) : CalendarItemViewHolder<EventCalendarItem>(composeView) {

    override fun bind(item: EventCalendarItem) {
        composeView.setContent {
            WhatsOnTheme {
                EventView(
                    modifier = Modifier
                        .fillMaxWidth()
                        .clickable {
                            eventSelectedListener.eventSelected(item)
                        },
                    event = item
                )
            }
        }
    }

}

fun Timestamp.format(formatter: DateTimeFormatter): String {
    return DateTime(millis).toString(formatter)
}
